File: client/android/core/push/src/main/java/app/mobiling/core/push/Push.kt
package app.mobiling.core.push

import app.mobiling.client.data.system.push.PushRegistrationGateway
import app.mobiling.client.usecase.system.push.PushRegisterUseCase

/**
 * Legacy-compatible Android entry point bridged to canonical system/push slices.
 */
class Push(
    baseUrl: String,
) {
    private val pushRegistrationGateway: PushRegistrationGateway = PushRegistrationGateway(baseUrl)
    private val pushRegisterUseCase: PushRegisterUseCase = PushRegisterUseCase(pushRegistrationGateway)

    fun register(
        token: String,
        platform: String = "android",
        appKey: String,
        deviceId: String,
        enabled: Boolean = true,
    ): Boolean = pushRegisterUseCase(
        token = token,
        platform = platform,
        appKey = appKey,
        deviceId = deviceId,
        enabled = enabled,
    )
}

File: client/ios/Sources/CorePush/Push.swift
import MobileClient
import Foundation

public struct Push {
  private let registerPushTokenUseCase: RegisterPushTokenUseCase
  private let pushTokenRegistrar: PushTokenRegistrar

  public init(baseUrl: String) {
    let registrar = PushTokenRegistrar(baseUrl: baseUrl)
    self.pushTokenRegistrar = registrar
    self.registerPushTokenUseCase = RegisterPushTokenUseCase(pushTokenRegistrar: registrar)
  }

  public func register(
    token: String,
    platform: String = "ios",
    appKey: String,
    deviceId: String,
    enabled: Bool = true
  ) async throws -> Bool {
    try await registerPushTokenUseCase(token: token, platform: platform, appKey: appKey, deviceId: deviceId, enabled: enabled)
  }

  public func registrationPayload(
    token: String,
    platform: String = "ios",
    appKey: String,
    deviceId: String,
    enabled: Bool = true
  ) -> PushRegistrationPayload {
    PushRegistrationPayload(token: token, platform: platform, appKey: appKey, deviceId: deviceId, enabled: enabled)
  }
}
